src/classes/BDI_DataImportFLSService.cls

```diff
@@ -137,7 +141,8 @@ public class BDI_DataImportFLSService {
                 && !field.getDescribe().isCreateable())
                 ||
                 (accessLevels.contains(AccessLevel.VALIDATE_UPDATE)
-                        && !field.getDescribe().isUpdateable())
+                        && !permissionsService.isUpdateable(field)
```

Comment:
Would it work to use the previously implemented `permissionsService.canUpdate(field.getDescribe(), false)` here?

Reply:
All of the "canUpdate" methods are static.  So we need an instance method to facilitate the test.  Re: naming and params of the instance method, I feel that the name "canUpdate" seems a bit like an unnecessary abstraction, and I prefer to pass the lighter-weight SObjectField when calling methods related to a field, rather than passing the describe object.  That being said, and instance method that follows the pattern you show here would be more in-line with the existing code.  Do you have a strong preference?  Happy to align it if you see more value that way.

Reply:
There are actually a few instance (non-static) versions of "canUpdate" already - give it a look (I think Jenny added these a while back for similar reasons as you are interested, aka supporting good unit test practices). I don't think the exact signature you are looking for is an instance method yet, so I think it's worth adding but I'd like to keep the naming consistent and also probably to use the FieldDescribe for the argument instead of the SObjectField just to be consistent with what's already in there, yeah

Reply:
(Check out line 350 of UTIL_Permissions! `public Boolean canUpdate(SObjectType sObjType, Set<SObjectField> sObjFields) {`)

Reply:
Don't use that signature, I do think you're right we need a variant that supports the FieldDescribe argument, maybe along with the boolean for throwing the exception, so we still are aligned with adding the new method

Reply:
Ok i see now, the instance methods are below the other static methods at the bottom of the class!  Ok I'll update this also - thanks!